taichi/program/launch_context_builder.cpp

```diff
@@ -77,7 +77,9 @@ void LaunchContextBuilder::set_ndarray_ptrs(int arg_id,
                                             uint64 data_ptr,
                                             uint64 grad_ptr) {
   set_struct_arg({arg_id, TypeFactory::DATA_PTR_POS_IN_NDARRAY}, data_ptr);
```

Comment:
Should we assert data_ptr and grad_ptr != 0 here

Reply:
SG! Will add it in the next PR!